hoodie-client/src/main/java/com/uber/hoodie/io/HoodieCommitArchiver.java

```diff
@@ -0,0 +1,273 @@
+
+    public static String dummyCommitTimeForArchive = "0";
+    private final Path archiveFilePath;
+    private final FileSystem fs;
+    private final HoodieWriteConfig config;
+    private HoodieLogFormat.Writer writer;
+
+    public HoodieCommitArchiver(HoodieWriteConfig config, FileSystem fs) {
+        this.fs = fs;
+        this.config = config;
+        this.archiveFilePath = HoodieArchivedTimeline
+            .getArchiveLogPath(config.getBasePath() + "/" + HoodieTableMetaClient.METAFOLDER_NAME);
+    }
+
+    private HoodieLogFormat.Writer openWriter(String commitTime) {
+        try {
+            return HoodieLogFormat.newWriterBuilder()
+                    .onParentPath(new Path(config.getBasePath()))
+                    .withFileId(archiveFilePath.getName())
+                    .withFileExtension(HoodieLogFile.DELTA_EXTENSION)
+                    .withFs(fs)
+                    .overBaseCommit(commitTime).build();
+        } catch(InterruptedException | IOException e) {
+            throw new HoodieException("Unable to initialize HoodieLogFormat writer", e);
+        }
+    }
+
+    private void close() {
+        try {
+            if(this.writer != null) {
+                this.writer.close();
+            }
+        } catch(IOException e) {
+            throw new HoodieException("Unable to close HoodieLogFormat writer", e);
+        }
+    }
+
+    /**
+     * Check if commits need to be archived. If yes, archive commits.
+     */
+    public boolean archiveIfRequired() {
+        try {
+            List<HoodieInstant> commitsToArchive = getCommitsToArchive().collect(Collectors.toList());
+            boolean success = true;
+            if (commitsToArchive.iterator().hasNext()) {
+                this.writer = openWriter(dummyCommitTimeForArchive);
+                log.info("Archiving commits " + commitsToArchive);
+                archive(commitsToArchive);
+                success = deleteInstants(commitsToArchive);
+            } else {
+                log.info("No Commits to archive");
+            }
+            return success & deleteOtherInstants();
+        } catch(Exception e) {
+            throw new HoodieException("Unable to archive commits", e);
+        } finally {
+            close();
+        }
+    }
+
+    private boolean deleteOtherInstants() {
+        // Delete clean and rollback files
+        List<HoodieInstant> toDelete = getOtherInstantsToDelete().collect(Collectors.toList());
+        if(!toDelete.isEmpty()) {
+            archive(toDelete);
+            log.info("Deleting actions " + toDelete);
+            return deleteInstants(toDelete);
+        }
+        return true;
+    }
+
+    private Stream<HoodieInstant> getOtherInstantsToDelete() {
+
+        int maxCommitsToKeep = config.getMaxCommitsToKeep();
+        int minCommitsToKeep = config.getMinCommitsToKeep();
+        HoodieTable table = HoodieTable.getHoodieTable(new HoodieTableMetaClient(fs, config.getBasePath(), true), config);
```

Comment:
Why not create table once and pass it in everywhere? you can refresh the timeline in the table